src/gui/properties/peerlistwidget.cpp

```diff
@@ -179,16 +182,20 @@ void PeerListWidget::showPeerListMenu(const QPoint&)
   QAction *act = menu.exec(QCursor::pos());
   if (act == 0) return;
   if (act == addPeerAct) {
-    BitTorrent::PeerAddress addr = PeerAdditionDlg::askForPeerAddress();
-    if (!addr.ip.isNull()) {
-      if (torrent->connectPeer(addr))
-        QMessageBox::information(0, tr("Peer addition"), tr("The peer was added to this torrent."));
-      else
-        QMessageBox::critical(0, tr("Peer addition"), tr("The peer could not be added to this torrent."));
-    }
-    else {
-      qDebug("No peer was added");
+    QList<BitTorrent::PeerAddress> peersList = PeersAdditionDlg::askForPeers();
+    int peerCount = 0;
+    foreach (const BitTorrent::PeerAddress &addr, peersList) {
+        if (torrent->connectPeer(addr)) {
+            qDebug("Adding peer %s...", qPrintable(addr.ip.toString()));
+            Logger::instance()->addMessage(tr("Manually adding peer %1...").arg(addr.ip.toString()));
+            peerCount++;
+        }
+        else {
+            QMessageBox::critical(0, tr("Peer addition"), tr("The peer %1 could not be added to this torrent.").arg(addr.ip.toString()));
```

Comment:
This could ugly if the user has pasted a bunch of bad IPs. I propose here you just leave a logger message of type warning. Also setting a flag eg `bool invalidPeers`. Then when outside the loop check the flag a post this MessageBox saying "Some peers couldn't be added. Check the Log for details."